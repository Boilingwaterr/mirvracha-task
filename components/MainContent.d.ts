import * as React from "react";
interface Props {
    list: [];
    listSize: number;
}
export declare const MainContent: React.NamedExoticComponent<Props>;
export {};
