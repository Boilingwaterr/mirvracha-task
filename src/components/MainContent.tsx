import * as React from "react";
import { useMemo, memo } from "react";

interface Props {
  list: [];
  listSize: number;
}

const MainContentComponent: React.FC<Props> = ({ list, listSize }) => {
  const countryList = useMemo(
    () =>
      list.map(({ name }, index) => {
        if (index <= listSize - 1) {
          return <p key={name}>{name}</p>;
        }
        return undefined;
      }),
    [list, listSize]
  );

  return <>{countryList}</>;
};

export const MainContent = memo(MainContentComponent);
