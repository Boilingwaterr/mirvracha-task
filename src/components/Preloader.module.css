.preloader {
  --bg-color: #22252b00;
  --size: 150px;
  --first-color: #ff7473;
  --second-color: #ffc952;
  --third-color: #47b8e0;
  --thickness: 3px;
  --duration: 3s;
  margin-top: 100px;
  background-color: var(--bg-color);
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: var(--duration);
}

.preloaderContent {
  position: relative;
  width: var(--size);
  height: var(--size);
}

.preloaderFirst,
.preloaderSecond,
.preloaderThird {
  position: absolute;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  animation: var(--duration) linear infinite;
}

.preloaderFirst {
  border-bottom: var(--thickness) solid var(--first-color);
  animation-name: first;
}

.preloaderSecond {
  border-right: var(--thickness) solid var(--second-color);
  animation-name: second;
}

.preloaderThird {
  border-top: var(--thickness) solid var(--third-color);
  animation-name: third;
}

@keyframes first {
  0% {
    transform: rotateX(35deg) rotateY(-45deg) rotateZ(0deg);
  }
  100% {
    transform: rotateX(35deg) rotateY(-45deg) rotateZ(360deg);
  }
}

@keyframes second {
  0% {
    transform: rotateX(50deg) rotateY(10deg) rotateZ(0deg);
  }
  100% {
    transform: rotateX(50deg) rotateY(10deg) rotateZ(360deg);
  }
}

@keyframes third {
  0% {
    transform: rotateX(35deg) rotateY(55deg) rotateZ(0deg);
  }
  100% {
    transform: rotateX(35deg) rotateY(55deg) rotateZ(360deg);
  }
}
