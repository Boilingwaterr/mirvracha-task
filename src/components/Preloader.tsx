import React from "react";
import s from "./Preloader.module.css";

const {
  preloaderContent,
  preloaderFirst,
  preloaderSecond,
  preloaderThird,
  preloader,
} = s;

export const Preloader = () => (
  <div className={preloader}>
    <div className={preloaderContent}>
      <div className={preloaderFirst}></div>
      <div className={preloaderSecond}></div>
      <div className={preloaderThird}></div>
    </div>
  </div>
);
