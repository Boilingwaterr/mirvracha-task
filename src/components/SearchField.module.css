.search_form {
  display: flex;
  height: 50px;
  width: 440px;
}

.search_field {
  display: flex;
  padding: 10px 5px;
  border: none;
  outline: none;
  background-color: #ececec;
  color: #3d3d3d;
  font-weight: bold;
  font-size: 20px;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  margin-right: 4px;
}
.drop_down {
  justify-content: center;
  align-items: center;
  font-size: 20px;
  color: #ececec;
  display: flex;
  background-color: brown;
  padding: 10px 40px;
  height: 100%;
  cursor: pointer;
  border-top-right-radius: 4px;
}
.drop_down_menu {
  font-size: 20px;
  color: #ececec;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #ca5e5e;
}
