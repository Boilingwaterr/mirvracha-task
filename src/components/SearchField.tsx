import * as React from "react";
import { useState, memo, useMemo } from "react";
import s from "./SearchField.module.css";

const { search_field, drop_down, search_form, drop_down_menu } = s;

interface Props {
  getCountriesByName: Function;
  setSize: Function;
  setInputValue: Function;
  inputValue: string;
}

const SearchFieldComponent: React.FC<Props> = ({
  getCountriesByName,
  setSize,
  setInputValue,
  inputValue,
}) => {
  function handleChange(e: React.ChangeEvent<HTMLInputElement>): void {
    setInputValue(e.target.value);
    getCountriesByName(e.target.value);
  }
  console.log(inputValue);
  return (
    <form className={search_form}>
      <input
        type="text"
        value={inputValue}
        onChange={handleChange}
        className={search_field}
      />

      <SizeList setSize={setSize} />
    </form>
  );
};

export const SearchField = memo(SearchFieldComponent);

const SizeList: React.FC<{ setSize: Function }> = memo(({ setSize }) => {
  const [isShowMenu, setIsShowMenu] = useState<boolean>(false);

  function handleChange(e: React.ChangeEvent<HTMLInputElement>): void {
    setSize(+e.target.value);
  }

  function handleMouseOver(): void {
    setIsShowMenu((prev: Boolean) => !prev);
  }

  const menuArr = ["10", "20", "50"];

  const menu = useMemo(
    () =>
      menuArr.map((num) => {
        return (
          <div key={`input-${num}`} className={drop_down_menu}>
            <input
              id={num}
              type="radio"
              value={num}
              name="size"
              onChange={handleChange}
            ></input>
            <label htmlFor={num}>{num}</label>
          </div>
        );
      }),
    [menuArr, handleChange]
  );

  return (
    <div onMouseEnter={handleMouseOver} onMouseLeave={handleMouseOver}>
      <div className={drop_down}>list size</div>
      <div>{isShowMenu ? menu : undefined}</div>
    </div>
  );
});
