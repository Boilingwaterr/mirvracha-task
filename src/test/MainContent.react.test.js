import React from "react";
import renderer from "react-test-renderer";

import { MainContent } from "../components/MainContent";

const list = [
  {
    name: "France",
  },
  {
    name: "Russian Federation",
  },
  {
    name: "The United States of America",
  },
];

const listSize = 10;

test("renders countries", () => {
  const component = renderer.create(
    <MainContent list={list} listSize={listSize} />
  );

  expect(component).toMatchSnapshot();
});
