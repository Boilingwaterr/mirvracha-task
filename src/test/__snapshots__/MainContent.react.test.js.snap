// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders countries 1`] = `
Array [
  <p>
    France
  </p>,
  <p>
    Russian Federation
  </p>,
  <p>
    The United States of America
  </p>,
]
`;
